packages/@lwc/template-compiler/src/codegen/helpers.ts

```diff
@@ -231,6 +232,8 @@ function isStaticNode(node: BaseElement): boolean {
 
     result &&= isElement(node);
 
+    result &&= nodeName !== HTML_TAG.SVG; // SVG contents must be parsed using parseSVGFragment
```

Comment:
I don't understand why we need to deoptimize `<svg>` elements, could you explain why this change is needed? 

[`parseSVGFragment`](https://github.com/salesforce/lwc/blob/3cb07f92c7b97c348db799665fda3efe9408678a/packages/@lwc/engine-core/src/framework/template.ts#L189-L190) works the same way as [`parseFragment`](https://github.com/salesforce/lwc/blob/3cb07f92c7b97c348db799665fda3efe9408678a/packages/@lwc/engine-core/src/framework/template.ts#L182) except that it wraps its content within an `<svg>` element to force the HTML parser to operate in the SVG namespace.

I got rid of this check locally and tests are still passing.

Reply:
You're right, this was a mistake I made while I was developing. Forgot that `parseSVGFragment` wasn't doing anything fancy.

Undoubtedly the perf is better with this line removed anyway, because an `<svg>` with many children wouldn't need to have all of its children get separate `<svg>` wrappers created on-the-fly.